File: CS/RangedMaskSample/MaskButtonEditMaskProperties.cs
using System;
using DevExpress.XtraEditors.Mask;
using DevExpress.Data.Mask;
using System.Globalization;

namespace RangedMaskSample
{
    class MaskButtonEditMaskProperties : TimeEditMaskProperties
    {
        private readonly MaskButtonEdit aEdit;

        public MaskButtonEditMaskProperties(MaskButtonEdit aedit)
            : base()
        {
            aEdit = aedit;
        }

        public virtual MaskManager CreatePatchedMaskManager()
        {
            CultureInfo managerCultureInfo = this.Culture;
            if (managerCultureInfo == null)
                managerCultureInfo = CultureInfo.CurrentCulture;

            string editMask = this.EditMask;
            if (editMask == null)
                editMask = string.Empty;

            return new MaskButtonEditMaskManager(editMask, false, managerCultureInfo, true, aEdit);
        }
    }
}

File: VB/RangedMaskSample/MaskButtonEditMaskProperties.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.XtraEditors.Mask
Imports DevExpress.Data.Mask
Imports System.Globalization

Namespace RangedMaskSample
	Friend Class MaskButtonEditMaskProperties
		Inherits TimeEditMaskProperties
		Private ReadOnly aEdit As MaskButtonEdit

		Public Sub New(ByVal aedit As MaskButtonEdit)
			MyBase.New()
			Me.aEdit = aedit
		End Sub

		Public Overridable Function CreatePatchedMaskManager() As MaskManager
			Dim managerCultureInfo As CultureInfo = Me.Culture
			If managerCultureInfo Is Nothing Then
				managerCultureInfo = CultureInfo.CurrentCulture
			End If

			Dim editMask As String = Me.EditMask
			If editMask Is Nothing Then
				editMask = String.Empty
			End If

			Return New MaskButtonEditMaskManager(editMask, False, managerCultureInfo, True, aEdit)
		End Function
	End Class
End Namespace
